File: backend/app/schemas/forecast.py
from pydantic import BaseModel, Field


class ForecastRequest(BaseModel):
    horizon_hours: int = Field(default=168, ge=24, le=168)
    start_from: str | None = None


class ForecastPoint(BaseModel):
    datetime: str
    predicted_mw: float
    lower_bound: float
    upper_bound: float


class ForecastResponse(BaseModel):
    forecast: list[ForecastPoint]
    horizon_hours: int
    model_used: str = "Prophet"

File: frontend/src/api/forecast.ts
import { api } from "./client";

export interface ForecastPoint {
  datetime: string;
  predicted_mw: number;
  lower_bound: number;
  upper_bound: number;
}

export interface ForecastResponse {
  forecast: ForecastPoint[];
  horizon_hours: number;
  model_used: string;
}

export async function fetchForecast(
  horizonHours: number = 168,
  startFrom?: string
): Promise<ForecastResponse> {
  const { data } = await api.post<ForecastResponse>("/forecast", {
    horizon_hours: horizonHours,
    start_from: startFrom,
  });
  return data;
}
